source/CMakeLists.txt

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required(VERSION 3.13)
+
+set(CMAKE_C_STANDARD 99)
+
+# Azure IoT FreeRTOS Middleware Library
+add_library(az_iot_middleware_freertos
+  ${CMAKE_CURRENT_LIST_DIR}/azure_iot_hub_client.c
+  ${CMAKE_CURRENT_LIST_DIR}/azure_iot_provisioning_client.c
+  ${CMAKE_CURRENT_LIST_DIR}/azure_iot.c
+)
+
+# The user needs to supply this
+target_link_libraries(az_iot_middleware_freertos
+  PUBLIC
+    middleware_config)
```

Comment:
this should be target_include_directories as config are headers?

Reply:
with PRIVATE

Reply:
It's built as an interface library which means it's only headers.

Reply:
will there be config.a or its all header?